File: src/Aneejian.Games.ClickMatch/Data/GameDto.cs
using System.ComponentModel.DataAnnotations;

namespace Aneejian.Games.ClickMatch.Data;

public class GameDto
{
	[Key]
	public int Id { get; set; }

	// foreign key for User ID
	public int UserId { get; set; }

	public UserDto User { get; set; } = null!;

	public int Level { get; set; }

	public int Score { get; set; }

	public double TimeTaken { get; set; }

	public bool GameWon { get; set; } = false;

	public bool GameLost { get; set; } = false;

	public bool GameAbandoned => !GameWon && !GameLost;
}

File: src/Aneejian.Games.ClickMatch/wwwroot/js/GameDTO.js

export class GameDTO {
    constructor(game) {
        this.id = game.id === 0 || null ? undefined : game.id;
        this.userId = game.userId;
        this.level = game.level;
        this.score = game.score;
        this.timeTaken = game.timeTaken;
        this.gameWon = game.gameWon;
        this.gameLost = game.gameLost;
        this.gameAbandoned = game.gameAbandoned;
    }
}
